zipline/assets/assets.py

```diff
@@ -27,112 +25,87 @@
 
 from zipline.errors import (
     ConsumeAssetMetaDataError,
-    InvalidAssetType,
     MultipleSymbolsFound,
     RootSymbolNotFound,
-    SidAssignmentError,
     SidNotFound,
     SymbolNotFound,
     MapAssetIdentifierIndexError,
 )
-from zipline.assets._assets import (
+from zipline.assets import (
     Asset, Equity, Future
 )
+from abc import ABCMeta
 
-log = Logger('assets.py')
-
-# Expected fields for an Asset's metadata
-ASSET_FIELDS = [
-    'sid',
-    'asset_type',
-    'symbol',
-    'root_symbol',
-    'asset_name',
-    'start_date',
-    'end_date',
-    'first_traded',
-    'exchange',
-    'notice_date',
-    'expiration_date',
-    'contract_multiplier',
-    # The following fields are for compatibility with other systems
-    'file_name',  # Used as symbol
-    'company_name',  # Used as asset_name
-    'start_date_nano',  # Used as start_date
-    'end_date_nano',  # Used as end_date
-]
-
-
-# Expected fields for an Asset's metadata
-ASSET_TABLE_FIELDS = [
-    'sid',
-    'symbol',
-    'asset_name',
-    'start_date',
-    'end_date',
-    'first_traded',
-    'exchange',
-]
-
-
-# Expected fields for an Asset's metadata
-FUTURE_TABLE_FIELDS = ASSET_TABLE_FIELDS + [
-    'root_symbol',
-    'notice_date',
-    'expiration_date',
-    'contract_multiplier',
-]
-
-EQUITY_TABLE_FIELDS = ASSET_TABLE_FIELDS
+from zipline.assets import (
+    NullAssetDBWriterLegacy,
+    AssetDBWriterLegacyFromList,
+    AssetDBWriterLegacyFromDictionary,
+    AssetDBWriterLegacyFromDataFrame,
+    AssetDBWriterLegacyFromReadable,
+    FUTURE_TABLE_FIELDS,
+    EQUITY_TABLE_FIELDS
+)
 
+log = Logger('assets.py')
 
 # Create the query once from the fields, so that the join is not done
 # repeatedly.
-FUTURE_BY_SID_QUERY = 'select {0} from futures where sid=?'.format(
+FUTURE_BY_SID_QUERY = 'select {0} from futures_contracts where sid=?'.format(
     ", ".join(FUTURE_TABLE_FIELDS))
 
 EQUITY_BY_SID_QUERY = 'select {0} from equities where sid=?'.format(
     ", ".join(EQUITY_TABLE_FIELDS))
 
 
+def get_relevant_writer(metadata):
```

Comment:
If this is creating the writers, should it be named create_relevant_writer? 'get_' makes me think that it is storing writers for the different types.

It may not be worth changing, depending on how soon we plan to deprecate this.